federation/pkg/federation-controller/util/events.go

```diff
@@ -0,0 +1,73 @@
+/*
```

Comment:
As a general rule, we are cutting down on our generic "util" packages and replacing them with more specific named ones.
How about naming this package eventsink? or federatedeventsink?